### PR TITLE
[FIX JENKINS-41642] Handle when favouriting a pipeline fails

### PR DESCRIPTION
**Pull request by tfennelly**
# Description

See [JENKINS-41642](https://issues.jenkins-ci.org/browse/JENKINS-41642).

Depends on https://github.com/jenkinsci/jenkins-design-language/pull/156

Simple error styled toast for when there's an favoriting error.

screenshot 2017-03-02 17 46 44

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
nice one @tfennelly ! toasty goodness. ATH happy. :bee:

**Comment by i386** [comment]
Got a screenshot handy?

**Comment by michaelneale** [comment]
can drop the status code

**Comment by tfennelly** [comment]
@i386 so you didn't look at the description? There's a screenshot there.

**Comment by kzantow** [comment]
It's his birthday, you expect him to look at the description?! ;)

**Comment by michaelneale** [comment]
Just finished a bottle of Moet.

File: blueocean-core-js/src/js/i18n/i18n.js [hunk 1]
```diff
@@ -164,7 +164,23 @@ export default function i18nTranslator(pluginName, namespace, onLoad) {
...
+                    }
+                    if (!passedParams.defaultValue) {
+                        passedParams.defaultValue = i18nKey;
+                    }
+                    if (logger.isDebugEnabled()) {
+                        logger.debug(`Normalized i18n key "${i18nKey}" to "${normalizedKey}".`);
+                    }
+                }
+                return fixedT(normalizedKey, passedParams);
+            };
```
Comment (tfennelly) [inline blueocean-core-js/src/js/i18n/i18n.js]: The above code is about making it a bit easier to just do the translation without having to define defaults etc i.e. just specify the default string and the key and default are inferred from that I think it's a bit better Vs having to define the `defaultValue` which, while not too bad, was a bit of a drag. See af675a1.

Comment (kzantow) [inline blueocean-core-js/src/js/i18n/i18n.js]: disagree about `defaultValue`, but big 👍  for just using the english as the key, I've said this all along

Comment (tfennelly) [inline blueocean-core-js/src/js/i18n/i18n.js]: Disagree in what way? You want it? Seems to contradict the second sentence.

Comment (kzantow) [inline blueocean-core-js/src/js/i18n/i18n.js]: Noooo! Disagree specifying a key _and_ english as a `defaultValue`, way too onerous.

Comment (kzantow) [inline blueocean-core-js/src/js/i18n/i18n.js]: it should have read: disagree with "while not too bad" ;)

Comment (tfennelly) [inline blueocean-core-js/src/js/i18n/i18n.js]: Ah stop ... not that big a deal ffs lol.

I think somewhere between my over-tolerance of things that others find to be troublesome and your intolerance of everything is probably a good balance 😉

**Comment by i386** [comment]
@tfennelly sorry to be a bother but one thing I did notice is that if you favourite something and it fails to do so the state of the favourite icon is the favourited one

**Pull request merged**